posts/2020-10-23-beta-20.0.0.12.adoc

```diff
@@ -0,0 +1,440 @@
+=== Support for MicroProfile JWT 1.2
+
+The MicroProfile JSON Web Token 1.2 specification allows using a JWT token for authenticating and authorizing requests to a service. The specification simplifies the configuration for managing the validation of the JWT by introducing new MicroProfile Config properties. Enhanced signature algorithm support is added in this Open Liberty beta.
+
+Version 1.2 of the MicroProfile JSON Web Token specification adds the following MicroProfile Config properties to control different aspects of the JWT validation.
+
+==== mp.jwt.token.header
+
+The `mp.jwt.token.header` property allows you to control the HTTP request header which is expected to contain a JWT token. You can specify either Authorization (default) or Cookie values.
+
+==== mp.jwt.token.cookie
+
+The `mp.jwt.token.cookie` property allows you to specify the name of the cookie which is expected to contain a JWT token. The default value is `Bearer` if not specified.
+
+==== mp.jwt.verify.audiences
+
+The `mp.jwt.verify.audiences` property allows you to create list of allowable audience (`aud`) values. At least one of these must be found in the claim. Previously, this had to be configured in the `server.xml` file. Now, you can configure the audiences in the MicroProfile Config property as follows:
+
+[source, java]
+----
+mp.jwt.verify.audiences=conferenceService
+----
+
+==== mp.jwt.verify.publickey.algorithm
+
+The `mp.jwt.verify.publickey.algorithm` property allows you to control the Public Key Signature Algorithm that is supported by the MicroProfile JWT endpoint. The default value is RSA256 if not specified. Previously, this had to be configured in the `server.xml` file. Now, you can configure the public key algorithm used for verification of the JWT in the MicroProfile Config property as follows:
+
+[source, java]
+----
+mp.jwt.verify.publickey.algorithm=ES256
+----
+
+The specification also adds support for the ES256 signature algorithm, while this Open Liberty beta supports using the RS384, RS512, HS384, HS512, ES256, ES384, and the ES512 signature algorithms.
+
+
+To enable the the MicroProfile JWT 1.2 feature in your app, add the following to your `server.xml`:
+
+[source, xml]
+----
+   <featureManager>
+       <feature>mpJwt-1.2</feature>
+   </featureManager>
+----
+
+For more information:
+
+link:https://github.com/eclipse/microprofile-jwt-auth/blob/master/spec/src/main/asciidoc/release-notes.asciidoc#release-notes-for-microprofile-jwt-12[Release Notes for MicroProfile JWT 1.2]
+
+link:https://github.com/eclipse/microprofile-jwt-auth/blob/master/spec/src/main/asciidoc/microprofile-jwt-auth-spec.asciidoc[MicroProfile JWT 1.2 Specification]
+
+[#lra]
+=== Support for MicroProfile LRA
+
+Long Running Actions (LRA) provide a cloud native transaction model based on a compensating Saga system. This allows loosely coupled services to coordinate long running activities, guaranteeing eventual data consistency without the need to put locks on data.
```

Comment:
Do we have a link to the compensating Saga system?  It might be nice for people (like me) who are not familiar with it - or for people who want more details.

Reply:
@yongja79 I too have no idea what the compensating saga model is.

Reply:
I've added this and had it default to the `#saga` section.